### PR TITLE
feat: rework validation

### PR DESCRIPTION
**Pull request by belopash**

### CODE REVIEW

File: src/manifest.ts [hunk 1]
```diff
@@ -1,146 +1,57 @@
...
+import { ManifestValue, ManifestProcessor, ManifestDeploymentConfig } from './types';
 
-export type ManifestProcessor = {
-  name: string;
-  cmd?: string[];
-  env?: Record<string, string>;
-};
-
-export type ManifestOptions = {
+export type ManifestParsiongOptions = {
```
Comment (mo4islona) [inline src/manifest.ts]: Typo

File: schemas/squid_manifest.json [hunk 2]
```diff
@@ -62,7 +62,7 @@
                             "minItems": 1,
                             "items": {
                                 "type": "string",
-                                "pattern": "^[:\\-\\/\\w.]+$"
+                                "pattern": "^([:\\-\\/\\w.]+|&&)$"
```
Comment (mo4islona) [inline schemas/squid_manifest.json]: That's not gonna work

**Review by mo4islona — CHANGES_REQUESTED** [review]

File: test/env.test.ts [hunk 3]
```diff
@@ -15,22 +15,18 @@ describe('Env', () => {
             OBJECT: {}
         processor:
           cmd: [ "node", "lib/processor" ]
-          env:
-            ARRAY: []
```
Comment (mo4islona) [inline test/env.test.ts:19]: Why did you remove processor checks?

File: test/env.test.ts [hunk 4]
```diff
@@ -50,15 +46,16 @@ describe('Env', () => {
...
       `);
 
-    expect(m.getErrors()).toEqual([
-      'Validation error occurred:',
-      '1) "deploy.processor[0].env.-invalid" is not allowed',
-      '2) "deploy.api.env.0invalid" is not allowed',
-    ]);
+    expect(error).toEqual(
+      new ManifestParsingError('Validation error occurred', [
+        '"deploy.processor.env.-invalid" is not allowed',
```
Comment (mo4islona) [inline test/env.test.ts:57]: Why is processor not an array now?

File: src/schema.ts [hunk 5]
```diff
@@ -104,14 +105,23 @@ export const manifestSchema = Joi.object({
...
-      .messages({
-        'array.unique': 'Processor names must be unique within a squid',
-      })
-      .min(1)
-      .required(),
+    migrate: Joi.object({
+      env: envSchema,
+      cmd: Joi.array().items(cmdSchema).min(1).required(),
+    })
+      .description('[DEPRECATED] Please use "init" instead')
```
Comment (mo4islona) [inline src/schema.ts]: Please use paths from root, i.e "deploy.init", to avoid confusing

File: test/env.test.ts (hunk 4 above)
Comment (belopash) [reply test/env.test.ts:57]: Because validation is done before transformation
